rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -72,17 +75,48 @@ def import_rmw_implementation():
 
     If :py:func:`select_rmw_implementation` has not previously been called then
     a default implementation will be selected implicitly before loading.
+
+    :raises: :py:class:`NoImplementationAvailableException` if there are no
+        implementations available.
     """
     global __rmw_implementation_module
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
+    available_implementations = get_rmw_implementations()
     if __selected_rmw_implementation is None:
         logger = logging.getLogger('rclpy')
-        select_rmw_implementation(get_rmw_implementations()[0])  # select the first one
+        if available_implementations:
+            select_rmw_implementation(available_implementations[0])  # select the first one
+        else:
+            raise NoImplementationAvailableException()
         logger.debug("Implicitly selecting the '{0}' rmw implementation."
                      .format(__selected_rmw_implementation))
     module_name = '._rclpy__{rmw_implementation}'.format(
         rmw_implementation=__selected_rmw_implementation,
     )
-    __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
+    try:
+        __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
+    except ImportError as exc:
+        if "No module named 'rclpy.{0}".format(module_name) in str(exc):
+            if available_implementations:
+                rmw_implementations_msg = '\n'.join(
+                    ['  - {0}'.format(x) for x in available_implementations]
+                )
+            else:
+                rmw_implementations_msg = '  No rmw implementation has a Python extension.'
+            log_args = [
+                __selected_rmw_implementation,
+                RCLPY_IMPLEMENTATION_ENV_NAME,
+                rmw_implementations_msg,
+            ]
+            # This message will only be printed if this exception goes unhandled.
+            add_unhandled_exception_addendum(
+                exc,
+                "\n"
+                "Failed to import the Python extension for the '{0}' rmw implementation.\n"
+                "A different rmw implementation can be selected using the '{1}' env variable.\n"
+                "These are the available rmw implementations:\n"
+                "{2}\n".format(*log_args)
```

Comment:
I guess it would be clearer, but this works for me. Since it's not a functional change, I'm not going to do it right now. Feel free to change it yourself.